### PR TITLE
Pre-Fill the primary company values in the form if they are mapped

### LINKED ISSUE #8400: No auto-fill of the COMPANY FIELDS in the Mautic form
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
When an existing contact with an assigned company gets to a Mautic landingpage, the data of the CONTACT FIELDS are automatically filled in the Mautic form. But not the assigned COMPANY FIELDS of this contact. The COMPANY FIELDS in the form are still blank. 

| Q   | A
| --- | ---
| Mautic version | v2.15.3
| PHP version | 7.2.21-1
| Browser | Edge, Firefox, Chrome, Safari

### Steps to reproduce
1. Create a landingpage with a form and the fields "contact name, company name and company street"
2. Create an email with a link to this landingpage
3. Send an email to an existing contact with an assigned company
4. The CONTACT FIELD "contact name" is correctly auto-filled. The COMPANY FIELDS "company name" and "company street" keep wrongly blank. (Because the data of the company name and company street already exist in the Mautic database)
 
### no Log errors

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

### PR DESCRIPTION
**Pull request by adiux**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | no
| New feature/enhancement? (use the a.x branch)      | yes
| Deprecations?                          | no
| BC breaks? (use the c.x branch)        | no
| Automated tests included?              | yes
| Related user documentation PR URL      | not needed
| Related developer documentation PR URL | not needed
| Issue(s) addressed                     | Fixes #8400, and forum discussion here https://forum.mautic.org/t/no-auto-fill-of-the-company-fields-in-the-mautic-form/12627/5

#### Description:

In a form, that has company fields mapped, those values are currently not pre-filled.

With this PR also the values from the **primary company** are pre-filled in the form.

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new form that has fields that map to contact and company fields (e.g. firstname, company address)
3. Make sure the pre-fill option is enabled on the field image
4. Add the form to a landingpage
5. Get a trackable link to this landingpage (so the contact is identified). E.g. by sending a personal email with the link to the landingpage (it has to be a html link, the URL as string in the body of the email does not generate a trackable link)
6. Test it without being logged in with Mautic (e.g. in Incognito)
7. The result should be something like this:
image

### CODE REVIEW

**Review by PatrickJenkner — APPROVED** [review]
Works exactly as described for a user.
Also, thank you for your detailed testing instructions. It makes things a lot easier for beginners like me :)

File: app/bundles/FormBundle/Model/FormModel.php [hunk 1]
```diff
@@ -871,8 +881,14 @@ public function populateValuesWithLead(Form $form, &$formHtml)
...
 
+        // get the contact (lead) and primary company field values
+        $leadArray = $this->primaryCompanyHelper->getProfileFieldsWithPrimaryCompany($lead);
+        if (!is_array($leadArray) || count($leadArray) <= 0) {
+            return;
+        }
+
         foreach ($autoFillFields as $field) {
-            $value = $lead->getFieldValue($field->getMappedField());
+            $value = isset($leadArray[$field->getMappedField()]) ? $leadArray[$field->getMappedField()] : '';
```
Comment (kuzmany) [inline app/bundles/FormBundle/Model/FormModel.php]: We can use shorten syntax

`$leadArray[$field->getMappedField()] ?? ''`

File: app/bundles/FormBundle/Model/FormModel.php [hunk 2]
```diff
@@ -109,6 +115,7 @@ public function __construct(
         ActionModel $formActionModel,
         FieldModel $formFieldModel,
         FormFieldHelper $fieldHelper,
+        PrimaryCompanyHelper $primaryCompanyHelper,
```
Comment (kuzmany) [inline app/bundles/FormBundle/Model/FormModel.php]: we can use from PHP8 `constructor property promotion` 

`private PrimaryCompanyHelper $primaryCompanyHelper` 

and you don't add anything else

**Review by kuzmany — DISMISSED** [review]
It works as you describe.
Just two small improvements in code syntax I left in comments

File: app/bundles/FormBundle/Model/FormModel.php (hunk 2 above)
Comment (adiux) [reply app/bundles/FormBundle/Model/FormModel.php]: love it! thanks!

**Review by kuzmany — APPROVED** [review]
Now looks good and could merge it 👍

**Comment by RCheesley** [comment]
Thanks folks! @adiux if you want this to be back-ported to 4.4.10 please make a PR on the 4.4 branch for testing! 🚀

**Pull request merged**